cpp/include/algorithms.h

```diff
@@ -365,4 +365,34 @@ void k_core(Graph* in_graph,
             gdf_column *vertex_id,
             gdf_column *core_number,
             Graph* out_graph);
+
+/**                                                                             
+ * @Synopsis  Get the maximal k for which a k-truss exists in the graph.
+ *                                                                              
+ * @Param[in] *in_graph              cuGRAPH graph descriptor with a valid edgeList or adjList
+ *                                                                              
+ * @Param[out] k_max                 Maximal k-truss found in graph.
+ *                                                                              
+ * @throws     cugraph::logic_error when an error occurs.
+ */                                                                             
+/* ----------------------------------------------------------------------------*/
+void k_truss_max(Graph *in_graph,
+                 int* k_max);
+
+/**                                                                             
+ * @Synopsis  Get the maximal k for which a k-truss exists in the graph.
+ *                                                                              
+ * @Param[in] *in_graph              cuGRAPH graph descriptor with a valid edgeList or adjList
```

Comment:
"with a valid edgeList or adjList" : does it work natively on both edgeList and adjList? Just checking because the C++ layer is not supposed to make decisions on format conversions. 
It is ok to accept only the format the CUDA kernels need at the C++ API level. Transformations should be done prior to reaching the C++ algorithm call by calling conversion functions from the python layer.

Reply:
Right now it's using adjlist (copying it from device to host) and using the CPU initialization process..
Ideally, it should use the edgelist and the GPU initialization process. The GPU initialization process is currently not working correctly, for this reason, we defaulted to the CPU initialization.